src/main/kotlin/io/UstMode1.kt

```diff
@@ -0,0 +1,111 @@
+            zip.file(trackFileName, Uint8Array(contentEncodedArray))
+        }
+        val option = JsZipOption().also { it.type = "blob" }
+        val blob = zip.generateAsync(option).await() as Blob
+        val name = project.name + ".zip"
+        val notifications = mutableListOf<ExportNotification>()
+        if (project.tempos.distinctBy { it.bpm }.count() > 1) {
+            notifications.add(ExportNotification.TempoChangeIgnored)
+        }
+        if (project.timeSignatures.any { it.numerator != DEFAULT_METER_HIGH || it.denominator != DEFAULT_METER_LOW }) {
+            notifications.add(ExportNotification.TimeSignatureIgnored)
+        }
+        if (features.contains(Feature.CONVERT_PITCH)) {
+            notifications.add(ExportNotification.PitchDataExported)
+        }
+        return ExportResult(blob, name, notifications)
+    }
+
+    private fun generateTrackContent(project: Project, track: Track, features: List<Feature>): String {
+        val builder = object {
+            var content = ""
+                private set
+
+            fun appendLine(line: String) {
+                content += "$line${LINE_SEPARATOR}"
+            }
+        }
+        builder.appendLine("[#VERSION]")
+        builder.appendLine("UST Version1.2")
+        builder.appendLine("[#SETTING]")
+        val bpm = project.tempos.first().bpm.toFixed(2)
+        builder.appendLine("Tempo=$bpm")
+        builder.appendLine("Tracks=1")
+        builder.appendLine("ProjectName=${track.name}")
+        var tickPos = 0L
+        var restCount = 0
+        var pitchData: UtauMode1TrackPitchData? = null
+        if (features.contains(Feature.CONVERT_PITCH))
+        {
+            pitchData = pitchToUtauMode1Track(track.pitch, track.notes)
+        }
+        for ((index, note) in track.notes.withIndex()) {
+            if (tickPos < note.tickOn) {
+                val restNoteNumber = (note.id + restCount).padStartZero(4)
+                builder.appendLine("[#$restNoteNumber]")
+                builder.appendLine("Length=${note.tickOn - tickPos}")
+                builder.appendLine("Lyric=R")
+                builder.appendLine("NoteNum=60")
+                restCount++
+            }
+            val noteNumber = (note.id + restCount).padStartZero(4)
+            builder.appendLine("[#$noteNumber]")
+            builder.appendLine("Length=${note.length}")
+            builder.appendLine("Lyric=${note.lyric}")
+            builder.appendLine("NoteNum=${note.key}")
+            builder.appendLine("PreUtterance=")
+            if (features.contains(Feature.CONVERT_PITCH)){
+                builder.appendLine("PBType=5")
+                val pitchString = makePitchDataString(pitchData?.notes?.get(index))
+                builder.appendLine("Piches=${pitchString}")
+                builder.appendLine("Pitches=${pitchString}")
+                builder.appendLine("PitchBend=${pitchString}")
```

Comment:
I think any of them should work?

Reply:
Yes, UTAU will read any of them for mode1's pitch data. But UTAU will save all these field all for its forward compatibility. So I follow it's behaviour here.

Reply:
I tried with my UTAU and it only saved `PitchBend`...